xarray/tests/duckarrays/base/strategies.py

```diff
@@ -0,0 +1,104 @@
+import hypothesis.extra.numpy as npst
+import hypothesis.strategies as st
+
+import xarray as xr
+
+
+def shapes(ndim):
+    return npst.array_shapes(min_dims=ndim, max_dims=ndim)
+
+
+dtypes = (
+    npst.integer_dtypes()
+    | npst.unsigned_integer_dtypes()
+    | npst.floating_dtypes()
+    | npst.complex_number_dtypes()
+)
+
+
+def numpy_array(shape=None):
+    if shape is None:
+        shape = shapes()
```

Comment:
```suggestion
def numpy_array(shape=npst.array_shapes()):
```

Strategies are safe default arguments (since they're not mutable), and based on the definition above it's an error to call `shapes()` without passing an `ndim` argument.